pallets/connectors-gateway/connectors-gateway-routers/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use cfg_traits::connectors::Router;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::dispatch::DispatchResult;
+use scale_info::TypeInfo;
+
+use crate::{axelar_evm::AxelarEVMRouter, ethereum_xcm::EthereumXCMRouter};
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub mod axelar_evm;
+pub mod axelar_xcm;
+pub mod ethereum_xcm;
+pub mod router;
+
+pub use axelar_evm::*;
+pub use axelar_xcm::*;
+pub use ethereum_xcm::*;
+
+type CurrencyIdOf<T> = <T as pallet_xcm_transactor::Config>::CurrencyId;
+type MessageOf<T> = <T as pallet_connectors_gateway::Config>::Message;
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+
+const CONNECTORS_FUNCTION_NAME: &str = "handle";
+const CONNECTORS_MESSAGE_PARAM: &str = "message";
+
+const AXELAR_FUNCTION_NAME: &str = "callContract";
+const AXELAR_DESTINATION_CHAIN_PARAM: &str = "destinationChain";
+const AXELAR_DESTINATION_CONTRACT_ADDRESS_PARAM: &str = "destinationContractAddress";
+const AXELAR_PAYLOAD_PARAM: &str = "payload";
+
+/// The routers used for outgoing messages.
+#[derive(Debug, Encode, Decode, Clone, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum DomainRouter<T>
+where
+	T: frame_system::Config
+		+ pallet_xcm_transactor::Config
+		+ pallet_connectors_gateway::Config
+		+ pallet_ethereum_transaction::Config
+		+ pallet_evm::Config,
+	T::AccountId: AsRef<[u8; 32]>,
+{
+	EthereumXCM(EthereumXCMRouter<T>),
+	AxelarEVM(AxelarEVMRouter<T>),
+	AxelarXCM(AxelarXCMRouter<T>),
+}
```

Comment:
I don't think we benefit from having this enum. What we want is a type that implements the `Router` trait. I see that we use this enum so that we offer this polymorpgic api with `do_init()` and `do_send()`, but the first is still unclear to me if we even want that instead of making that a governance proposal setting that initial state or even a migration, and the latter is just a duplicate of `Router::send`.

Reply:
I think that this enum will come in handy when setting a domain router - polkadotJS, for example, will present all 3 available options which IMO would make things easier.

I'm still not sold on the idea of having a separate step for setting the initial state, especially since not all routers require an initial step, we would then have to over-complicate and over-document the steps for specific routers.

Reply:
Ah yeah you are absolutely right, I forgot about that part 🤦🏻 and about the second bit, not a blocker for me, if you think that's the way to go let's do that